pkg/xds/experimental/client/listener.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+package client
+
+import (
+	"log/slog"
+
+	"github.com/cilium/cilium/pkg/envoy/xds"
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+// listener is a helper structure for watchers focused on handling a single, registered callback.
+type listener struct {
+	ID      uint64
+	typeUrl string
+	cb      WatcherCallback
+	trigger chan uint64
+
+	log       *slog.Logger
+	resources xds.ResourceSource
+
+	lastVersion uint64
+	mux         lock.Mutex
+	nextVersion uint64
+}
+
+var _ xds.ResourceVersionObserver = (*listener)(nil)
```

Comment:
// listener implements xds.ResourceVersionObserver

Reply:
Added the requested comment